### PR TITLE
Update to Scintilla 5.3.3 and Lexilla 5.2.2

### PR DESCRIPTION
**Pull request by chcg**
update to https://www.scintilla.org/scintilla533.zip with:

    Released 8 February 2023.
    Fix SCI_LINESJOIN bug where carriage returns were incorrectly retained. Bug #2372.
    Fix SCI_VERTICALCENTRECARET to update the vertical scroll position.
    When an autocompletion list is shown in response to SCN_CHARADDED, do not process character as fill-up or stop. This avoids closing immediately when a character may both trigger and finish autocompletion.
    On Cocoa fix character input bug where dotless 'i' and some other extended Latin characters could not be entered. The change also stops SCI_ASSIGNCMDKEY from working with these characters on Cocoa. Bug #2374.
    On GTK, support IME context. Feature #1476.
    On GTK on Win32, fix scrolling speed to not be too fast. Bug #2375.
    On Qt, fix indicator drawing past left of text pane over margin. Bug #2373, Bug #1956.
    On Qt, allow scrolling with mouse wheel when scroll bar hidden.

and https://www.scintilla.org/lexilla522.zip with

    Released 8 February 2023.
    C++: Fix keywords that start with non-ASCII. Also affects other lexers. Issue #130.
    Matlab: Include more prefix and suffix characters in numeric literals. Issue #120.
    Matlab: More accurate treatment of line ends inside strings. Matlab and Octave are different here. Issue #18.
    Modula-3: Don't treat identifier suffix that matches keyword as keyword. Issue #129.
    Modula-3: Fix endless loop in folder. Issue #128.
    Modula-3: Fix access to lines beyond document end in folder. Issue #131.
    Python: Don't highlight match and case as keywords in contexts where they probably aren't used as keywords. Pull request #122.
    X12: Support empty envelopes. Bug #2369.

- update CMakeLists.txt to latest changes within vcxproj file

### CODE REVIEW

**Comment by Yaron10** [comment]
@chcg,

https://github.com/notepad-plus-plus/notepad-plus-plus/issues/12219.

Thank you.

**Comment by chcg** [comment]
@Yaron10 Thanks for finding this merge issue. Fixed.

**Comment by Yaron10** [comment]
@chcg,

Great. 👍 
Thank you.

File: PowerEditor/src/CMakeLists.txt [hunk 1]
```diff
@@ -385,13 +385,13 @@ IF (WIN32)
 if ( MSVC )
 	#do not use for mingw builds
 	SET(CMAKE_CXX_FLAGS "/EHa /MP /W4")
-	SET(defs -DUNICODE -D_UNICODE -D_WIN32_WINNT=0x0600 -D_USE_64BIT_TIME_T -DTIXML_USE_STL -DTIXMLA_USE_STL -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_CRT_NON_CONFORMING_SWPRINTFS )
+	SET(defs -DUNICODE -D_UNICODE -D_WIN32_WINNT=_WIN32_WINNT_VISTA -D_USE_64BIT_TIME_T -DTIXML_USE_STL -DTIXMLA_USE_STL -DNOMINMAX -DOEMRESOURCE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_CRT_NON_CONFORMING_SWPRINTFS -D_CRT_NON_CONFORMING_WCSTOK -D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING )
```
Comment (ozone10) [inline PowerEditor/src/CMakeLists.txt:388]: @chcg 
Just a question. Are both `-D_CRT_NON_CONFORMING_SWPRINTFS` and `-D_CRT_NON_CONFORMING_WCSTOK` needed? Notepad++ msvc builds don't use these macros anymore.

Comment (chcg) [reply PowerEditor/src/CMakeLists.txt]: @ozone10 Thanks. No, should be inline with the settings of the visual studio build. Corrected.

**Pull request closed without merging**